File: benchmarks/fib_tco46.c
int fibTco(int n, int a, int b) {
  if (n == 0) {
    return a;
  } else {
    if (n == 1) {
      return b;
    } else {
      return fibTco(n - 1, b, a + b);
    }
  }
}

int main(int argc, char const *argv[]) {
  fibTco(46, 0, 1);
  return 0;
}

File: benchmarks/fib_tco46.rb
def fib_tco(n, a, b)
  if (n == 0):
    a
  else
    if (n == 1):
      b
    else
      fib_tco(n - 1, b, a + b)
    end
  end
end

fib_tco(46, 0, 1)